Automattic/woocommerce-payments
# Fix FedEx shipping multi-currency rates

Fixes #10022 

#### Changes proposed in this Pull Request

Originally, we had discovered the inconsistency with shipping rates only happened with JPY. It turns out it wasn't because JPY is a zero decimal currency, but because of the value after conversion.

The MCCY compatibility works by changing the WooCommerce store currency on the fly, but the FedEx API gets the converted values of the product in absolute terms, so if it surpasses a certain threshold, it applies insurance to it.

The way we pass the actual store currency in `set_settings`, but with the price already converted, makes it so that the FedEx API adds insurance to the shipping rates.

To avoid this problem, we're now preventing the price from being converted so that the calculations can be made correctly.

#### Testing instructions

Follow instructions to set up the Woo FedEx shipping extension: https://github.com/Automattic/woocommerce-payments/pull/3131

**Test: Ensure shipping rates are the same in different currencies**

1. As a merchant, set the store currency to USD.
2. Create a shippable product with dimensions and weight (It can be 5 for everything).
3. As a shopper, add the shippable product to the cart.
4. Enter a US address to the cart, e.g `San Francisco / CA / 94110`.
5. Observe the shipping costs in USD.
6. Open a new tab and switch the store currency to JPY using the currency switcher.
7. Ensure the FedEx shipping costs in JPY respect the converted amount in USD – To double check, you can use the conversion preview from **WooCommerce > Settings > Multi-currency > JPY**.

**Regression test: insurance remains for higher-ticket products when paying with USD**

1. As a merchant, change the shippable product price to 1000 USD.
2. As a shopper, switch the currency to USD and notice the shipping rates are slightly higher due to insurance.
3. As a merchant, navigate to **WooCommerce > Settings > Shipping** and edit the shipping zone with FedEx as a shipping method. 
4. Change the currency to JPY using the currency switcher and observe the shipping rates are equivalent from step 2 when converted.
5. Click "Edit" from FedEx, uncheck "Enable Insurance" and save the changes.
6. As a shopper, repeat step 2 or refresh the tab and notice there's no insurance added to the shipping rates.

-------------------


**Post merge**
